src/App.jsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import './App.css';
+import { Goodlist } from './components/GoodList/GoodList';
+
+const url = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`
+
+class App extends React.Component {
+  state = {
+    goods: [],
+  }
+
+  getAll = async () => {
```

Comment:
alternative solution: you fetch all goods always, write them to the state -> then the corresponding method show those set of goods you need in each of the cases

in this case you won't need to make request every time